### PR TITLE
Make test cluster code an "manufacturer extension cluster"

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
Spec does not define a test cluster, however we need it to test things. Existing code makes it look like a "standard cluster",

#16886 asks for manufacturer specific clusters - this makes the test cluster one of those.

#### Change overview
Change code 0x050F to 0xFFF1050F to match a "test MC 0xFFF1 prefix" according to chip spec.

#### Testing
CI will validate that test cluster tests still pass.

### CODE REVIEW

File: examples/tv-casting-app/tv-casting-common/tv-casting-app.matter [hunk 1]
```diff
@@ -1854,7 +1854,7 @@ client cluster TargetNavigator = 1285 {
   command NavigateTarget(NavigateTargetRequest): NavigateTargetResponse = 0;
 }
 
-server cluster TestCluster = 1295 {
+server cluster TestCluster = 4293985551 {
```
Comment (bzbarsky-apple) [inline examples/tv-casting-app/tv-casting-common/tv-casting-app.matter:1857]: Why is this even implementing TestCluster?  Please file a followup to get that fixed and ping @chrisdecenzo about it?

File: src/app/util/util.h [hunk 2]
```diff
@@ -74,8 +74,7 @@
 // Cluster name structure
 typedef struct
 {
-    uint16_t id;
-    uint16_t mfgCode;
+    uint32_t id;
```
Comment (bzbarsky-apple) [inline src/app/util/util.h:77]: Shouldn't this be `ClusterId`?

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/tv-casting-app/tv-casting-common/tv-casting-app.matter (hunk 1 above)
Comment (andy31415) [reply examples/tv-casting-app/tv-casting-common/tv-casting-app.matter:1857]: https://github.com/project-chip/connectedhomeip/issues/19172 flagged this one as well. It has a lot of TestCluster errors

File: src/app/util/util.h (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/util/util.h:77]: My point is that assuming that "cluster id" is "uint32_t" is bad and we should not do it.

Comment (andy31415) [reply src/app/util/util.h:77]: Ah, data-type wise? Sure, that I believe I can change (assuming includes work)

**Review by tcarmelveilleux — APPROVED** [review]

**Pull request merged**